src/Protocol/Diaspora.php

```diff
@@ -2329,15 +2352,20 @@ private static function addReshareActivity(array $item, int $parent_message_id,
 	 */
 	private static function receiveReshare(array $importer, SimpleXMLElement $data, string $xml, int $direction): bool
 	{
-		$author = XML::unescape($data->author);
+		$author = WebFingerUri::fromString(XML::unescape($data->author));
 		$guid = XML::unescape($data->guid);
 		$created_at = DateTimeFormat::utc(XML::unescape($data->created_at));
-		$root_author = XML::unescape($data->root_author);
+		try {
+			$root_author = WebFingerUri::fromString(XML::unescape($data->root_author));
+		} catch (\InvalidArgumentException $e) {
+			return false;
```

Comment:
Logging?

Reply:
There was none before, not sure it would be helpful.